### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by shnhrrsn**
Supersedes #11 

This greatly simplifies the logic of the plugin + adds eval tests to make sure the code actually runs!

### CODE REVIEW

**Comment by MaxMEllon** [comment]
Truly Thank you! 😄

File: package.json [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
-  "main": "lib",
+  "main": "src",
   "scripts": {
-    "compile": "babel -V && babel src --out-dir lib",
-    "pretest": "npm run compile & npm run lint",
+    "pretest": "npm run lint",
     "test": "ava",
-    "lint": "xo --ignore lib/*.js",
-    "fix": "xo --ignore lib/*.js --fix"
+    "lint": "xo --ignore src/*.js",
```
Comment (MaxMEllon) [inline package.json]: Please lint to files contain `src/`.

```diff
--- a/package.json
+++ b/package.json
@@ -6,8 +6,8 @@
   "scripts": {
     "pretest": "npm run lint",
     "test": "ava",
-    "lint": "xo --ignore src/*.js",
-    "fix": "xo --ignore src/*.js --fix"
+    "lint": "xo",
+    "fix": "xo --fix"
```

**Comment by shnhrrsn** [comment]
@MaxMEllon I had to drop Node 0.x from Travis to get tests pass as it looks like Ava [no longer supports 0.x](https://travis-ci.org/MaxMEllon/babel-plugin-transform-isNil/jobs/286352643)

**Review by MaxMEllon — CHANGES_REQUESTED** [review]
Seriously thank you :heart_eyes:

**Comment by MaxMEllon** [comment]
> drop Node 0.x 

I think there is no problem, because other famous project drop Node 0.x.

**Review by MaxMEllon — APPROVED** [review]
LGTM

**Comment by MaxMEllon** [comment]
Published

**Comment by shnhrrsn** [comment]
Awesome, thanks for the quick turnaround!

**Pull request merged**